app/src/androidTest/java/com/github/dedis/student20_pop/ui/CameraPermFragmentTest.java

```diff
@@ -0,0 +1,62 @@
+package com.github.dedis.student20_pop.ui;
+
+import android.Manifest;
+import android.content.pm.PackageManager;
+
+import androidx.core.app.ActivityCompat;
+import androidx.test.ext.junit.rules.ActivityScenarioRule;
+
+import com.github.dedis.student20_pop.MainActivity;
+import com.github.dedis.student20_pop.R;
+
+import org.junit.Rule;
+import org.junit.Test;
+
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.action.ViewActions.click;
+import static androidx.test.espresso.assertion.ViewAssertions.matches;
+import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+
+/**
+ * Class handling camera permission fragment tests
+ */
+public class CameraPermFragmentTest {
+
+    @Rule public ActivityScenarioRule<MainActivity> rule = new ActivityScenarioRule<>(MainActivity.class);
+
+    @Test
+    public void checkCorrectTabIsDisplayed() {
+        // Check that the permission is not granted
+        rule.getScenario().onActivity(activity ->
+                assertThat(ActivityCompat.checkSelfPermission(activity, Manifest.permission.CAMERA),
+                        not(is(PackageManager.PERMISSION_GRANTED))));
+
+        onView(withId(R.id.tab_connect)).perform(click());
+        onView(withId(R.id.fragment_camera_perm)).check(matches(isDisplayed()));
+    }
+
+    /*@Test
+    public void grantPermissionChangeTab() {
+        // Check that the permission is not granted
+        rule.getScenario().onActivity(activity ->
+                assertThat(ActivityCompat.checkSelfPermission(activity, Manifest.permission.CAMERA),
+                        not(is(PackageManager.PERMISSION_GRANTED))));
+
+        onView(withId(R.id.tab_connect)).perform(click());
+        onView(withId(R.id.fragment_camera_perm)).check(matches(isDisplayed()));
+
+        onView(withId(R.id.allow_camera_button)).perform(click());
+        //This is very unstable but I don't see how to do that in an other way. So if the button isn't found, the test just exits
+        onView(withText("Allow")).perform(click()); // Not working
+
+        rule.getScenario().onActivity(activity ->
+                assertThat(ActivityCompat.checkSelfPermission(activity, Manifest.permission.CAMERA),
+                        is(PackageManager.PERMISSION_GRANTED)));
+
+        onView(withId(R.id.fragment_connect)).check(matches(isDisplayed()));
+    }*/
+}
```

Comment:
Do you want to keep this commented test ? Otherwise you can remove it before merging

Reply:
I'm trying to figure out a way to test this. That's why I kept it.

Reply:
I removed it